compiler/resolution/functionResolution.cpp

```diff
@@ -8322,11 +8323,13 @@ static void resolveExports() {
 
       resolveSignatureAndFunction(fn);
 
-      exps.push_back(fn);
+      if (isLibrary) { exps.push_back(fn); }
     }
   }
 
-  fixupExportedFunctions(exps);
+  if (isLibrary) {
+    fixupExportedFunctions(exps);
+  }
```

Comment:
Note that this doesn't protect other calls to fixupExportedFunctions from being run outside of `--library` compilation (whether or not that happens today).  Though it means additional work occurs, I think it would be better to put this limitation in fixupExportedFunctions itself.

Reply:
That seems reasonable to me!